diff --git a/src/provisioner.ts b/src/provisioner.ts
index c1568af..28a44c5 100644
--- a/src/provisioner.ts
+++ b/src/provisioner.ts
@@ -99,8 +99,9 @@
         this.pendingRequests.set(channelId, approveFn);
         setTimeout(() => approveFn(false, true), timeout);
 
-        await channel.send(`${requestor} on matrix would like to bridge this channel. Someone with permission` +
-            " to manage webhooks please reply with `!matrix approve` or `!matrix deny` in the next 5 minutes");
+        // await channel.send(`${requestor} on matrix would like to bridge this channel. Someone with permission` +
+        //     " to manage webhooks please reply with `!matrix approve` or `!matrix deny` in the next 5 minutes");
+        approveFn(true);
         return await deferP;
 
     }
