diff --git a/src/bot.ts b/src/bot.ts
index 8bc73d4..1e6ea67 100644
--- a/src/bot.ts
+++ b/src/bot.ts
@@ -568,7 +568,8 @@ export class DiscordBot {
             }
             const link = `https://discord.com/channels/${chan.guild.id}/${chan.id}/${editEventId}`;
             embedSet.messageEmbed.description = `[Edit](${link}): ${embedSet.messageEmbed.description}`;
-            await this.send(embedSet, opts, roomLookup, event);
+            log.warn("not editing sent Matrix -> Discord message");
+            // await this.send(embedSet, opts, roomLookup, event);
         } catch (err) {
             // throw wrapError(err, Unstable.ForeignNetworkError, "Couldn't edit message");
             log.warn(`Failed to edit message ${event.event_id}`);
