diff --git a/configs/default_hooks/sxmo_hook_apps.sh b/configs/default_hooks/sxmo_hook_apps.sh
index ba70a31..f1ce0d5 100755
--- a/configs/default_hooks/sxmo_hook_apps.sh
+++ b/configs/default_hooks/sxmo_hook_apps.sh
@@ -31,6 +31,7 @@ write_line_app audacity "$icon_mic Audacity" "audacity"
 write_line_app gnome-calculator "$icon_clc Calculator" "gnome-calculator"
 write_line_app calcurse "$icon_clk Calcurse" "sxmo_terminal.sh calcurse"
 write_line_app cmus "$icon_mus Cmus" "sxmo_terminal.sh cmus"
+write_line_app cozy "$icon_mus Cozy" "cozy"
 write_line_app dino "$icon_msg Dino" "GDK_SCALE=1 dino"
 write_line_app dolphin "$icon_dir Dolphin" "dolphin"
 write_line_app emacs "$icon_edt Emacs (Terminal)" "sxmo_terminal.sh emacs -nw"
@@ -62,8 +63,10 @@ write_line_app ii "$icon_msg Ii" "sxmo_terminal.sh ii"
 write_line_app ipython "$icon_trm IPython" "sxmo_terminal.sh ipython"
 write_line_app kasts "$icon_rss Kasts" "kasts"
 write_line_app kmail "$icon_eml KMail" "kmail"
+write_line_app komikku "$icon_bok Komikku" "komikku"
 write_line_app kontact "$icon_msg Kontact" "kontact"
 write_line_app konversation "$icon_msg Konversation" "konversation"
+write_line_app koreader "$icon_bok KOReader" "koreader"
 write_line_app kwrite "$icon_edt Kwrite" "kwrite"
 write_line_app lagrange "$icon_glb Lagrange" "lagrange"
 write_line_app lf "$icon_dir Lf" "sxmo_terminal.sh lf"
